### PR TITLE
GH-466 - user module: refactor public accessors into functions and rename functions

### LINKED ISSUE #466: user module: refactor public accessors into functions and rename functions
Originating from #272 and discussed in Nov 29 refinement.

- Refactor public accessors into functions.
- If needed, rename functions to be short and into context
i.e. productService.get() instead of productService.getProduct()
Avoid redundancy. 
Additionally we are allowed to not mirror the commerce platform.

There is a follow up ticket for facades to load once in the store (epic/loadonceinstore).

### PR DESCRIPTION
**Pull request by znikola**
Closes #466

### CODE REVIEW

File: projects/storefrontlib/src/lib/user/facade/user.service.ts [hunk 1]
```diff
@@ -1,78 +1,56 @@
...
-  );
-
   constructor(private store: Store<fromStore.UserState>) {}
 
-  loadUserDetails(userId: string) {
-    this.store.dispatch(new fromStore.LoadUserDetails(userId));
+  /**
+   * Returns a user
+   */
+  getDetails(): Observable<User> {
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/lib/user/facade/user.service.ts]: Can we just use `get()`? I am not sure about it.

Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: Done

File: projects/storefrontlib/src/lib/user/facade/user.service.ts [hunk 2]
```diff
@@ -1,78 +1,56 @@
...
+    return this.store.pipe(select(fromStore.getDetails));
   }
 
-  loadTitles() {
-    this.store.dispatch(new fromStore.LoadTitles());
+  /**
+   * Loads the user's details
+   */
+  loadDetails(userId: string): void {
```
[the remark this answers is not in the document]
Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: Done

File: projects/storefrontlib/src/lib/user/facade/user.service.ts [hunk 3]
```diff
@@ -84,19 +62,20 @@ export class UserService {
...
   }
 
-  getCountry(isocode: string): Observable<any> {
-    return this.store.pipe(select(fromStore.countrySelectorFactory(isocode)));
-  }
-
-  loadDeliveryCountries() {
-    this.store.dispatch(new fromStore.LoadDeliveryCountries());
+  /**
+   * Returns order details
```
[the remark this answers is not in the document]
Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: I don't understand this comment?

File: projects/storefrontlib/src/lib/user/facade/user.service.ts [hunk 4]
```diff
@@ -105,31 +84,55 @@ export class UserService {
...
+  /**
+   * Returns a loaded flag for order history list
+   */
+  getOrderHistoryListLoaded(): Observable<boolean> {
+    return this.store.pipe(select(fromStore.getOrdersLoaded));
   }
 
-  loadPaymentMethods(userId: string) {
+  /**
+   * Retrieves a payment method
```
[the remark this answers is not in the document]
Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: Good catch

File: projects/storefrontlib/src/lib/user/facade/user.service.ts [hunk 5]
```diff
@@ -105,31 +84,55 @@ export class UserService {
...
+   * @param userId a user ID
+   */
+  loadPaymentMethods(userId: string): void {
     this.store.dispatch(new fromStore.LoadUserPaymentMethods(userId));
   }
 
-  setPaymentMethodAsDefault(userId: string, paymentMethodId: string) {
+  /**
+   * Returns payment methods
```
[the remark this answers is not in the document]
Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: I don't understand this comment?

File: projects/storefrontlib/src/lib/user/facade/user.service.ts [hunk 6]
```diff
@@ -138,7 +141,13 @@ export class UserService {
...
   }
 
-  deleteUserPaymentMethod(userId: string, paymentMethodId: string) {
+  /**
+   * Deletes the payment method
+   *
+   * @param userId a user ID
+   * @param paymentMethodId a payment method ID
+   */
+  deleteUserPaymentMethod(userId: string, paymentMethodId: string): void {
```
[the remark this answers is not in the document]
Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: Done

File: projects/storefrontlib/src/lib/user/facade/user.service.ts [hunk 7]
```diff
@@ -147,7 +156,99 @@ export class UserService {
...
+   * Retrieves delivery countries
+   */
+  loadDeliveryCountries(): void {
+    this.store.dispatch(new fromStore.LoadDeliveryCountries());
+  }
+
+  /**
+   * Returns all delivery countries
+   */
+  getAllDeliveryCountries(): Observable<Country[]> {
```
[the remark this answers is not in the document]
Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: Done

File: projects/storefrontlib/src/lib/user/facade/user.service.ts [hunk 8]
```diff
@@ -147,7 +156,99 @@ export class UserService {
...
+   * @param countryIsoCode
+   */
+  loadRegions(countryIsoCode: string): void {
+    this.store.dispatch(new fromStore.LoadRegions(countryIsoCode));
+  }
+
+  /**
+   * Returns all regions
+   */
+  getAllRegions(): Observable<Region[]> {
```
[the remark this answers is not in the document]
Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: Done

File: projects/storefrontlib/src/lib/user/facade/user.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: The comment is "Returns an order details". It function only return **one** order data.

File: projects/storefrontlib/src/lib/user/facade/user.service.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: In the doc comment, we say: return user's all payment methods.

File: projects/storefrontlib/src/lib/user/facade/user.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: Done

File: projects/storefrontlib/src/lib/user/facade/user.service.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (znikola) [reply projects/storefrontlib/src/lib/user/facade/user.service.ts]: Done

**Pull request merged**